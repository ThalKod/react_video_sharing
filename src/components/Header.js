import React from 'react';

export default class Header extends React.Component{
  render(){
    return (
        <div>Hello Header !</div>
    )
  }
}
