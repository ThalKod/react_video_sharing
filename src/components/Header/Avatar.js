import React from "react";

export default () => {
  const style = {
    avatar:{
      backgroundColor: "#a2aec1",
      width: 50,
      height: 50,
      borderRadius: 50/2,
      textAlign: "center",

    },
    text: {
      color: "white",
      fontSize: 15,
      marginTop: 5,
    }
  };
  return (
        <div style={style.avatar} className="avatar pull-left">
          <div style={style.text}>
            TM
          </div>
        </div>
  )
};
