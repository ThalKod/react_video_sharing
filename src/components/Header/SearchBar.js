import React from "react";

export default class SearchBar extends React.Component {
  render(){
    return(
        <div className="topsearch">
          <div className="input-group">
            <span className="input-group-addon" id="sizing-addon2"><i className="fa fa-search"></i></span>
            <input type="text" className="form-control" placeholder="Search" aria-describedby="sizing-addon2"/>
          </div>
        </div>
    )
  }
}
