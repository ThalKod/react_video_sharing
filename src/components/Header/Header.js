import React from 'react';
import { Link } from "react-router-dom"
import { connect } from "react-redux";

import SearchBar from "./SearchBar";
import Avatar from "./Avatar";
import bulbLight from "../../assets/images/icon_bulb_light.png";
import logo from "../../assets/images/logo.svg";

class Header extends React.Component{

  state = {
    loggedIn: false
  };

  componentDidMount = () => {
    const { token } = this.props;
    if(token) {
      this.setState({ loggedIn: true});
    }
  };

  componentDidUpdate = (prevProps) => {
    const { token } = this.props;
    if(prevProps.token !== token){
      this.setState( prevStates => ({
        loggedIn: !prevStates.loggedIn
      }))
    }
  };

  render() {
    const { loggedIn } = this.state;
    return (
        <div className="container-fluid">
          <div className="row">
            <div className="btn-color-toggle">
              <img src={bulbLight} alt="light bulb"/>
            </div>
            <div className="navbar-container">
              <div className="container">
                <div className="row">
                  <div className="col-lg-3 col-sm-2 col-xs-2">
                    <Link to="/" className="navbar-brand">
                      <img src={logo} alt="Circle Logo" className="logo"/>
                    </Link>
                  </div>
                  <div className="visible-xs visible-sm clearfix"/>
                  <div className="col-lg-6 col-sm-6 col-xs-12">
                    <SearchBar/>
                  </div>
                  <div className="visible-xs clearfix"/>
                  <div className="col-lg-2 col-sm-4  col-xs-8">
                    <div className="pull-right">
                      {loggedIn ? <Avatar/> :
                          <div className="loginsignup">
                            <Link to="/signin">Login</Link> . <Link to="/signup">Signup</Link>
                          </div>
                      }
                    </div>
                    <div className="clearfix"/>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
    )
  }
}

const mapStateToProps = (state) => ({
  token: state.auth.userToken,
});

export default connect(mapStateToProps)(Header);
