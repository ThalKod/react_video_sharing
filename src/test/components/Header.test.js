import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import { Header } from "components/Header/Header";

describe("Header", () => {

  const token = "test token";
  const getMyInfo = (callback) => {
    callback({error: false})
  };
  const username = "John Doe";

  it("should render the signin and logout link when not login", () => {
    const wrapper = shallow(<Header />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render the Avatar and signout button if login", () => {

    const wrapper = shallow(<Header token={token} getMyInfo={getMyInfo} username={username} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(
        wrapper.state("loggedIn")
    ).toBe(true);
  });

  it("should update the header when login", () => {
    const wrapper = shallow(<Header />);
    wrapper.setProps({ token });
    expect(
        wrapper.state("loggedIn")
    ).toBe(true);
  });

  it("should update the header when signout", () => {

    const wrapper = shallow(<Header token={token} getMyInfo={getMyInfo} username={username} />);
    wrapper.setProps({ token: "" });
    expect(
        wrapper.state("loggedIn")
    ).toBe(false);
  })
});


