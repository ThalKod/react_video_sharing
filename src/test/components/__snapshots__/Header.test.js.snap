// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the Avatar and signout button if login 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="btn-color-toggle"
    >
      <img
        alt="light bulb"
        src="icon_bulb_light.png"
      />
    </div>
    <div
      className="navbar-container"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-lg-3 col-sm-2 col-xs-2"
          >
            <Link
              className="navbar-brand"
              replace={false}
              to="/"
            >
              <img
                alt="Circle Logo"
                className="logo"
                src="logo.svg"
              />
            </Link>
          </div>
          <div
            className="visible-xs visible-sm clearfix"
          />
          <div
            className="col-lg-6 col-sm-6 col-xs-12"
          >
            <SearchBar
              submit={[Function]}
            />
          </div>
          <div
            className="visible-xs clearfix"
          />
          <div
            className="col-lg-2 col-sm-4  col-xs-8"
          >
            <div
              className="pull-right"
            >
              <div
                className="flex"
              >
                <Link
                  replace={false}
                  to="/channel/undefined"
                >
                  <_default
                    username="John Doe"
                  />
                </Link>
                <button
                  className="unStyledButton"
                  type="submit"
                >
                  Sign Out
                </button>
                <Link
                  className="upload-button"
                  replace={false}
                  to="/upload"
                >
                  <i
                    className="cv cvicon-cv-upload-video"
                  />
                </Link>
              </div>
            </div>
            <div
              className="clearfix"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Header should render the signin and logout link when not login 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="btn-color-toggle"
    >
      <img
        alt="light bulb"
        src="icon_bulb_light.png"
      />
    </div>
    <div
      className="navbar-container"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-lg-3 col-sm-2 col-xs-2"
          >
            <Link
              className="navbar-brand"
              replace={false}
              to="/"
            >
              <img
                alt="Circle Logo"
                className="logo"
                src="logo.svg"
              />
            </Link>
          </div>
          <div
            className="visible-xs visible-sm clearfix"
          />
          <div
            className="col-lg-6 col-sm-6 col-xs-12"
          >
            <SearchBar
              submit={[Function]}
            />
          </div>
          <div
            className="visible-xs clearfix"
          />
          <div
            className="col-lg-2 col-sm-4  col-xs-8"
          >
            <div
              className="pull-right"
            >
              <div
                className="loginsignup"
              >
                <Link
                  replace={false}
                  to="/signin"
                >
                  Login
                </Link>
                 . 
                <Link
                  replace={false}
                  to="/signup"
                >
                  Signup
                </Link>
              </div>
            </div>
            <div
              className="clearfix"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
